ufl/algorithms/map_integrands.py

```diff
@@ -19,20 +19,24 @@
 from ufl.constantvalue import Zero
 
 
-def map_integrands(function, form, only_integral_type=None):
+def map_integrands(function, form, only_integral_type=None,
+                   reconstruct_integral_domain=False):
     """Apply transform(expression) to each integrand
     expression in form, or to form if it is an Expr.
     """
     if isinstance(form, Form):
-        mapped_integrals = [map_integrands(function, itg, only_integral_type)
+        mapped_integrals = [map_integrands(function, itg, only_integral_type,
+                                           reconstruct_integral_domain)
                             for itg in form.integrals()]
         nonzero_integrals = [itg for itg in mapped_integrals
                              if not isinstance(itg.integrand(), Zero)]
         return Form(nonzero_integrals)
     elif isinstance(form, Integral):
         itg = form
         if (only_integral_type is None) or (itg.integral_type() in only_integral_type):
-            return itg.reconstruct(function(itg.integrand()))
+            integrand = function(itg.integrand())
+            domain = integrand.ufl_domain() if reconstruct_integral_domain else None
```

Comment:
Arguably this reconstruction of the domain is always the right thing to do?

Reply:
I agree. It just resulted in the test failure. If I fix that then that's what I'll do.